pkg/asset/manifests/vsphere/cloudproviderconfig.go

```diff
@@ -27,7 +27,10 @@ func CloudProviderConfig(clusterName string, p *vspheretypes.Platform) (string,
 	printIfNotEmpty(buf, "server", p.VCenter)
 	printIfNotEmpty(buf, "datacenter", p.Datacenter)
 	printIfNotEmpty(buf, "default-datastore", p.DefaultDatastore)
-	printIfNotEmpty(buf, "folder", clusterName)
+	printIfNotEmpty(buf, "folder", p.Folder)
+	if p.Folder == "" {
+		printIfNotEmpty(buf, "folder", clusterName)
+	}
```

Comment:
why do we have 2 prints for the same field? wouldn't that be incorrect??
```
folder = ""
folder = "clustername"
```

Reply:
@abhinavdahiya this is what we were talking about at the end of grooming.
If Folder is empty we will continue to use clusterName.

Reply:
`printIfNotEmpty` checks if the value is ""